packages/trader/src/Modules/Contract/Components/LastDigitPrediction/last-digit-prediction.tsx

```diff
@@ -93,9 +111,9 @@ const LastDigitPrediction = ({
     // latest last digit refers to digit and spot values from latest price
     // latest contract digit refers to digit and spot values from last digit contract in contracts array
     const latest_tick_pip_size = tick ? +tick.pip_size : null;
-    const latest_tick_ask_price = tick && tick.ask ? tick.ask.toFixed(latest_tick_pip_size) : null;
-    const latest_tick_digit = latest_tick_ask_price ? +latest_tick_ask_price.split('').pop() : null;
-    const position = tick ? getOffset()[latest_tick_digit] : getOffset()[last_contract_digit.digit];
+    const latest_tick_ask_price = tick && tick.ask ? tick.ask.toFixed(latest_tick_pip_size ?? undefined) : null;
```

Comment:
The nullish coalescing (`??`) operator is a logical operator that returns its right-hand side operand when its left-hand side operand is `null` or `undefined`, and otherwise returns its left-hand side operand. But in line above if the right-hand side operand is `null` or `undefined` you also return `undefined`. Was it planned?

Reply:
(method) Number.toFixed(fractionDigits?: number | undefined): string

**tofixed** method expects `(number | undefined)` as inputs and returns a `string`. Since `latest_tick_pip_size` is `number | null` by inference. I had to stuff `undefined` with this method to satisfy the typescript error.

Reply:
Oh, I see, thank you for explanation